README.md

```diff
@@ -18,68 +18,83 @@ As a part of the bridge, a Kafka producer will be responsible for producing mess
 
 The ToMaR is a set of patterns the user provides defining how the MQTT Bridge maps MQTT topic names to Kafka topic names.
-A Mapping Rule is a model that contains an `MQTT topic pattern` and a `Kafka topic template`. 
+A Mapping Rule is a model that contains an `MQTT topic pattern`, a `Kafka topic template`, and optionally a `Kafka record key`.
 All the incoming MQTT message's topic should match an `MQTT topic pattern` in the ToMaR so that the bridge knows in which Kafka topic to produce this message.
 This Kafka topic is defined by a template, `Kafka Topic template`.
 However, if the incoming MQTT message's topic does not match any pattern in the ToMaR, the Bridge has a default Kafka topic where the incoming message will be mapped to, known as `messages_default`.
 
-A valid ToMaR is a JSON file that contains an array of mapping rules. 
+The optional `Kafka record key` is used to define the key of the Kafka record that will be produced to the Kafka topic.
+It is defined by a template as well, and its default value is `null`.
+A valid ToMaR is a JSON file that contains an array of mapping rules.
 Each mapping rule is a JSON object that contains two properties: `mqttTopic` and `kafkaTopic`.
-The following is an example of a valid TOMAR:
+The following is an example of a valid ToMaR:
 
 ```json
 [
   {
-    "mqttTopic": "building/{building}/room/{room}/#",
-    "kafkaTopic": "building_{building}_room_{room}"
+     "mqttTopic": "building/(\\w+)/room/(\\d{1,4}).*",
+     "kafkaTopic": "building_$1",
+     "kafkaKey": "room_$2"
   },
   {
-    "mqttTopic": "sensors/+/data",
+    "mqttTopic": "sensors/([^/]+)/data",
     "kafkaTopic": "sensor_data"
   },
   {
-    "mqttTopic": "sensors/#",
-    "kafkaTopic": "sensor_others"
+     "mqttTopic": "sensors.*",
+     "kafkaTopic": "sensor_others"
+  },
+  {
+    "mqttTopic": "devices/([^/]+)/data/(\b(all|new|old)\b)",
+     "kafkaTopic": "device_$1_data",
+     "kafkaKey": "device_$2"
   }
 ]
 ```
 
-The wildcard `#` represents one or more levels in the MQTT topic hierarchy.
-The wildcard `+` represents a single level in the MQTT topic hierarchy.
+The expression `.*` is used to represent the wildcard `#`, which in turn represents one or more levels in the MQTT topic hierarchy.
+The expression `([^/]+)` is used to represent the wildcard `+`, which in turn represents a single level in the MQTT topic hierarchy.
 It worth's mentioning that it is the user's responsibility to adhere to the [MQTT 3.1.1 naming conventions](http://docs.oasis-open.org/mqtt/mqtt/v3.1.1/os/mqtt-v3.1.1-os.html#_Toc398718106) when defining the MQTT topic patterns.
 
-The parts with curly braces in the MQTT topic pattern are called `placeholders`.
-The MQTT Bridge will extract the values of these placeholders from the MQTT topic and use them to construct the Kafka topic name using the `kafkaTopic` template.
+Placeholders in the `KafkaTopic` and `kafkaKey` templates are defined using the `$` character followed by a number. 
```

Comment:
```suggestion
Placeholders in the `kafkaTopic` and `kafkaKey` templates are defined using the `$` character followed by a number. 
```

Reply:
I have added this to my other changes to avoid having 1 PR to each change.